### PR TITLE
[1LP][RFR]Rectify create_rest method for rsa credentials

### PR DESCRIPTION
**Pull request by valaparthvi**
Changes introduced with the PR:
1. Rectify `_fill_rsa_endpoint_dicts` method so that it creates the provider correctly via REST and works for v2v testing.
2. Modify `_fill_default_endpoint_dicts` method.

{{ pytest: cfme/tests/cloud_infra_common/test_rest_providers.py::test_create_provider --use-template-cache -sqvvv }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 20d46f46685ad46ea83c33bbe7aa03563025d022:

`cfme/common/provider.py`:
- :bangbang: Line 377:27: **E231** *missing whitespace after ':'*
- :bangbang: Line 378:21: **E123** *closing bracket does not match indentation of opening bracket's line*
- :bangbang: Line 381:31: **E231** *missing whitespace after ':'*
- :bangbang: Line 382:21: **E123** *closing bracket does not match indentation of opening bracket's line*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit d96d5d85e0aa16cc560c2fafc2d997b347ec3d2b:

`cfme/common/provider.py`:
- :bangbang: Line 318:70: **E222** *multiple spaces after operator*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/common/provider.py [hunk 1]
```diff
@@ -310,22 +310,18 @@ def _fill_default_endpoint_dicts(self, provider_attributes, connection_configs):
...
-            connection_configs.append(default_connection)
         else:
             raise AssertionError("Provider wasn't added. "
                 "No credentials info found for provider {}.".format(self.name))
 
-        cert = getattr(endpoint_default, "ca_certs", None)
-        if cert and self.appliance.version >= "5.8":
-            default_connection["endpoint"]["certificate_authority"] = cert
-            connection_configs.append(default_connection)
+        if getattr(endpoint_default, "ca_certs", None):
```
Comment (john-dupuy) [inline cfme/common/provider.py]: Optional: as far as I can tell `getattr` is being used in the same was as `hasattr` throughout this file, so why not remain consistent?

Comment (digitronik) [reply cfme/common/provider.py]: I think a better option is `hasattr` as we are just checking.

**Pull request merged**